test/e2e/page-objects/pages/settings/shield/shield-detail-page.ts

```diff
@@ -301,4 +309,27 @@ export default class ShieldDetailPage {
 
     await this.checkPaymentMethod(paymentMethod);
   }
+
+  /**
+   * Click on the payment method button to open the payment modal
+   */
+  async clickPaymentMethod(): Promise<void> {
+    console.log('Clicking on payment method button');
+    await this.driver.waitForSelector(this.paymentMethodButton);
+    await this.driver.clickElement(this.paymentMethodButton);
+    await this.driver.waitForSelector(this.shieldPaymentModal);
+  }
+
+  /**
+   * Select payment method token in the payment modal
+   *
+   * @param paymentMethodText - The full payment method text to select (e.g., 'Pay with USDT', 'Pay with USDC')
+   */
+  async selectPaymentMethodInModal(paymentMethodText: string): Promise<void> {
+    console.log(`Selecting payment method: ${paymentMethodText}`);
+    await this.driver.clickElement({
+      css: this.shieldPaymentMethodTokenButton,
+      text: paymentMethodText,
```

Comment:
instead of defining a selector here, could we move this on top?

ie 

```
 private readonly paymentMethodSelection = (paymentMethodText: string) => {
    return {...    };
  };
````